distutils/dist.py

```diff
@@ -354,7 +354,12 @@ def _gen_paths(self):
         prefix = '.' * (os.name == 'posix')
         filename = prefix + 'pydistutils.cfg'
         if self.want_user_cfg:
-            yield pathlib.Path('~').expanduser() / filename
+            try:
+                user_home = pathlib.Path('~').expanduser()
+            except RuntimeError:
+                self.announce("Failed to locate user home directory. Skipping user config.", logging.WARNING)
+            else:
+                yield user_home / filename
```

Comment:
Hello (disclaimer: not a distutils maintainer).

Would it make sense to use a proper warning here?

```suggestion
            try:
                yield pathlib.Path('~').expanduser() / filename
            except RuntimeError:
                warnings.warn("Failed to locate user home directory. Skipping user config.")
```

Reply:
Maybe! I'm certainly open to other error handling approaches, so long as a failure to locate the home directory doesn't cause failure. Let's see what a maintainer has to say.

Reply:
I don't have an instinct on the matter, so I'd defer to abravalheri's instinct. Let's go with that.